9 post new comment

MartinSWDev opened the pull request.

Commit:
add functionality to add comment with basic error handling

Commit:
remove console log

leekli reviewed (changes requested):
Hi Martin, well done so far on this ticket - it's not quite finished, comments below. Well done on the styling it looks really nice :)

Ticket-specific comments:
* The form doesn't reset after I submit a comment (think about controlled components, and how a text field should reset/clear after submission)
* I can see you give the user a confirmation message, but I can't see my comment. I have to refresh the page to be able to see my comment (which interestingly also logs me out so I cannot post another comment - more on this in the next set of comments), maybe have a think about optimistically rendering the comment to the top of the pile so the user can instantly see it along with a confirmation message?

I've had a good play around with your app, i've picked up on a few things worth thinking about - these aren't ticket specific to this ticket number 9, and I wouldn't fail the ticket because of these, but they're worth thinking about/looking into later:
* As a user, when I am on the homepage I see a login button, but it doesn't do anything when I click it which confused me... I then saw a dropdown box at the top called 'categories' which when dropped down shows a list of usernames, I then worked out I click one of those and then press login which worked... 2 suggestions: Re-name that categories title of the dropdown box, and put it right above the login button so it's obviously it's a thing which exists together
* If I refresh the page, it forgets who I am logged in as, so if I try to post a comment on another article (had to refresh as I couldn't see a posted comment to check it actually worked and added the comment) I now can't because I'm not logged in, I couldn't find any easy navigation to take me back to the login page so I had to manually change the url address bar to take me back to the homepage... we wouldn't expect a user to do this, so have a think about how you can help users with this.
* The navigation panel which expands out to show all categories is at the bottom right hand corner of the app in mobile mode (using Safari)... is this a design choice or is my browser rendering it not how you want? A nav panel would usually be at the top or on the left hand side on most websites. It's not actually an issue, but it's worth thinking about where users usually expect to find certain things on websites.
* I tried go to an article which doesn't exist (800), and it gave me a blank page and just the header bar at the top, I'd amend this to present them with a 404 error page

Hope that helps :)

Commit:
improve ux from login to post comment

MartinSWDev commented:
> Hi Martin, well done so far on this ticket - it's not quite finished, comments below. Well done on the styling it looks really nice :)
> 
> Ticket-specific comments:
> 
>     * The form doesn't reset after I submit a comment (think about controlled components, and how a text field should reset/clear after submission)
> 
>     * I can see you give the user a confirmation message, but I can't see my comment. I have to refresh the page to be able to see my comment (which interestingly also logs me out so I cannot post another comment - more on this in the next set of comments), maybe have a think about optimistically rendering the comment to the top of the pile so the user can instantly see it along with a confirmation message?
> 
> 
> I've had a good play around with your app, i've picked up on a few things worth thinking about - these aren't ticket specific to this ticket number 9, and I wouldn't fail the ticket because of these, but they're worth thinking about/looking into later:
> 
>     * As a user, when I am on the homepage I see a login button, but it doesn't do anything when I click it which confused me... I then saw a dropdown box at the top called 'categories' which when dropped down shows a list of usernames, I then worked out I click one of those and then press login which worked... 2 suggestions: Re-name that categories title of the dropdown box, and put it right above the login button so it's obviously it's a thing which exists together
> 
>     * If I refresh the page, it forgets who I am logged in as, so if I try to post a comment on another article (had to refresh as I couldn't see a posted comment to check it actually worked and added the comment) I now can't because I'm not logged in, I couldn't find any easy navigation to take me back to the login page so I had to manually change the url address bar to take me back to the homepage... we wouldn't expect a user to do this, so have a think about how you can help users with this.
> 
>     * The navigation panel which expands out to show all categories is at the bottom right hand corner of the app in mobile mode (using Safari)... is this a design choice or is my browser rendering it not how you want? A nav panel would usually be at the top or on the left hand side on most websites. It's not actually an issue, but it's worth thinking about where users usually expect to find certain things on websites.
> 
>     * I tried go to an article which doesn't exist (800), and it gave me a blank page and just the header bar at the top, I'd amend this to present them with a 404 error page
> 
> 
> Hope that helps :)

Thanks Lee,

In reverse order:

- 404 error was hiding behind the header so have moved it down
- yes menu is bottom right on mobile (i havent done  anythinf for web css yet) this is in the "thumb zone" where all the buttons should be for mobile
- have added a login button to nav to navigate back to login
- have renamed and moved the login dropdown
- comment now shows on page when posted
- text area now clears
